### PR TITLE
#51 gitlab gRPC logic moved to gitaly package.

### PR DESCRIPTION
**Pull request by stepanvalyavskiy**
#51 `/gitaly/server` package contains registration and start up logic for `gitaly` specific gRPC and provides `start` & `stop` interfaces.

### CODE REVIEW

File: gitaly/server/server.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	degitxpb.RegisterSmartHTTPServiceServer(grpcServer, &degitxpb.UnimplementedSmartHTTPServiceServer{})
+	degitxpb.RegisterWikiServiceServer(grpcServer, &degitxpb.UnimplementedWikiServiceServer{})
+	degitxpb.RegisterConflictsServiceServer(grpcServer, &degitxpb.UnimplementedConflictsServiceServer{})
+	degitxpb.RegisterRemoteServiceServer(grpcServer, &degitxpb.UnimplementedRemoteServiceServer{})
+	degitxpb.RegisterServerServiceServer(grpcServer, server.NewServer(logger, version.GetVersion()))
+	degitxpb.RegisterObjectPoolServiceServer(grpcServer, &degitxpb.UnimplementedObjectPoolServiceServer{})
+	degitxpb.RegisterHookServiceServer(grpcServer, &degitxpb.UnimplementedHookServiceServer{})
+	degitxpb.RegisterInternalGitalyServer(grpcServer, &degitxpb.UnimplementedInternalGitalyServer{})
+
+	reflection.Register(grpcServer)
```
Comment (g4s8) [inline gitaly/server/server.go:57]: @stepanvalyavskiy I think we can move all these registration lines to some method, e.g. `func NewGrpcServer() Server` to create and register gRPC services and return server with `Start()` method

File: gitaly/server/server.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	}
+	s.server = grpcServer
+	return nil
+}
+
+func (s *GrpcServer) Stop() {
+	if s.server != nil {
+		s.server.Stop()
+	}
+}
```
Comment (g4s8) [inline gitaly/server/server.go:79]: @stepanvalyavskiy what is the use case of this `Stop()` method? I'd suggest starting this server asynchronously and use `context` package primitives for cancellation, see https://golang.org/pkg/context/

**Review by g4s8 — APPROVED** [review]
@stepanvalyavskiy thanks. Please check few minor comments

File: gitaly/server/server.go (hunk 1 above)
Comment (stepanvalyavskiy) [reply gitaly/server/server.go:57]: In Gitaly it calls `RegisterAll`

**Review by g4s8 — APPROVED** [review]
@stepanvalyavskiy thanks

**Pull request merged**